gestore/tests/test_utils.py

```diff
@@ -0,0 +1,144 @@
+import pkg_resources
+from unittest.mock import call, patch
+
+from django.contrib.auth.models import User
+from django.test import TestCase
+
+from demoapp.factories.django import UserFactory
+from gestore import utils
+
+
+class TestUtils(TestCase):
+    def test_get_pip_packages(self):
+        packages = utils.get_pip_packages()
+        self.assertIsInstance(packages, dict)
+
+        # Get packages should match all the libraries fetched
+        # from pkg_resources
+        for package in pkg_resources.working_set:
+            self.assertEqual(
+                packages.pop(package.project_name),
+                package.version
+            )
+
+        # No more packages should be left after the popping thing above
+        self.assertEqual(len(packages), 0)
+
+    def test_has_conflict(self):
+        user = UserFactory()
+        self.assertTrue(utils.has_conflict(User, user.id))
+        self.assertFalse(utils.has_conflict(User, 99999))
+
+    def test_get_obj_from_str(self):
+        user = UserFactory()
+
+        # Make sure we get the right object
+        representation = 'auth.User.%s' % user.id
+        fetched_object = utils.get_obj_from_str(representation)
+        self.assertEqual(user, fetched_object)
+
+        # Test object in DB
+        representation = 'auth.User.99999999999'
+        with self.assertRaises(User.DoesNotExist):
+            utils.get_obj_from_str(representation)
+
+        # Test wrong representation
+        representation = 'auth.User'
+        with self.assertRaisesMessage(
+                ValueError, 'not enough values to unpack'
+        ):
+            utils.get_obj_from_str(representation)
+
+        representation = 'auth.something.User.someid'
+        with self.assertRaisesMessage(
+                ValueError, 'too many values to unpack'
+        ):
+            utils.get_obj_from_str(representation)
+
+
+class TestGetStrFromModel(TestCase):
+    def setUp(self) -> None:
+        self.obj = UserFactory()
+
+    def test_no_object_id(self):
+        path = utils.get_str_from_model(User)
+        self.assertEqual(path, 'auth.User')
+
+    def test_with_object_id(self):
+        path = utils.get_str_from_model(User, object_id=123)
+        self.assertEqual(path, 'auth.User.123')
+
+
+class TestUtilsWritePackagesDiff(TestCase):
+    def setUp(self) -> None:
+        self.local = {
+            'a': '1.2.3',
+            'b': '2.3.4',
+            'c': '3.4.5',
+        }
+
+    @patch('builtins.print')
+    def test_packages_equal(self, mock_print):
```

Comment:
**Nit:** `@patch('builtins.print')` seems a bit odd and could break. Pytest `capsys` is the go-to tool in such cases.

```suggestion
    def test_packages_equal(self, capsys):
```